File: web/frontend/lib/core/rbac_constants.dart
class Permissions {
  static const chatRead = 'chat.read';
  static const chatSend = 'chat.send';
  
  static const canvasView = 'canvas.view';
  
  static const memoryRead = 'memory.read';
  static const memoryWrite = 'memory.write';
  
  static const skillsList = 'skills.list';
  static const skillsInstall = 'skills.install';
  static const skillsManage = 'skills.manage';
  
  static const cronsList = 'crons.list';
  static const cronsManage = 'crons.manage';
  
  static const terminalExecSafe = 'terminal.exec.safe';
  static const terminalExecStandard = 'terminal.exec.standard';
  static const terminalExecPrivileged = 'terminal.exec.privileged';
  
  static const settingsRead = 'settings.read';
  static const settingsAdmin = 'settings.admin';
  
  static const governanceView = 'governance.view';
  static const governanceResolve = 'governance.resolve';
  
  static const acpSpawn = 'acp.spawn';
  static const acpManage = 'acp.manage';
  
  static const usersList = 'users.list';
  static const usersManage = 'users.manage';
  
  static const auditRead = 'audit.read';

  static const List<String> guestPermissions = [
    chatRead,
    canvasView,
    memoryRead,
    skillsList,
    cronsList,
    settingsRead,
    governanceView,
    terminalExecSafe,
  ];

  static const List<String> userPermissions = [
    chatSend,
    memoryWrite,
    skillsInstall,
    cronsManage,
    terminalExecStandard,
    governanceResolve,
    acpSpawn,
  ];

  static const List<String> adminPermissions = [
    skillsManage,
    terminalExecPrivileged,
    settingsAdmin,
    acpManage,
    usersList,
    usersManage,
    auditRead,
  ];
}

class Roles {
  static const guest = 'guest';
  static const user = 'user';
  static const admin = 'admin';
  static const superadmin = 'superadmin';
}

File: app/web/src/lib/utils/rbac-constants.ts
/**
 * RBAC constants — 1:1 port of core/rbac_constants.dart
 */

export const Permissions = {
  chatRead: 'chat.read',
  chatSend: 'chat.send',
  canvasView: 'canvas.view',
  memoryRead: 'memory.read',
  memoryWrite: 'memory.write',
  skillsList: 'skills.list',
  skillsInstall: 'skills.install',
  skillsManage: 'skills.manage',
  cronsList: 'crons.list',
  cronsManage: 'crons.manage',
  terminalExecSafe: 'terminal.exec.safe',
  terminalExecStandard: 'terminal.exec.standard',
  terminalExecPrivileged: 'terminal.exec.privileged',
  settingsRead: 'settings.read',
  settingsAdmin: 'settings.admin',
  governanceView: 'governance.view',
  governanceResolve: 'governance.resolve',
  acpSpawn: 'acp.spawn',
  acpManage: 'acp.manage',
  usersList: 'users.list',
  usersManage: 'users.manage',
  auditRead: 'audit.read',
} as const;

export const guestPermissions = [
  Permissions.chatRead,
  Permissions.chatSend,
  Permissions.canvasView,
  Permissions.memoryRead,
  Permissions.skillsList,
  Permissions.cronsList,
  Permissions.settingsRead,
  Permissions.governanceView,
];

export const userPermissions = [
  Permissions.memoryWrite,
  Permissions.skillsInstall,
  Permissions.cronsManage,
  Permissions.terminalExecSafe,
  Permissions.terminalExecStandard,
  Permissions.governanceResolve,
  Permissions.acpSpawn,
];

export const adminPermissions = [
  Permissions.skillsManage,
  Permissions.terminalExecPrivileged,
  Permissions.settingsAdmin,
  Permissions.acpManage,
  Permissions.usersList,
  Permissions.usersManage,
  Permissions.auditRead,
];

export const Roles = {
  guest: 'guest',
  user: 'user',
  admin: 'admin',
  superadmin: 'superadmin',
} as const;
